tests/api/test_flows.py

```diff
@@ -514,6 +516,36 @@ async def test_create_flow_persists_serialized_flow(self, project_id, flow):
         # confirm the keys in the serialized flow match the form we'd expect
         assert persisted_flow.serialized_flow == flow.serialize()
 
+    async def test_create_flow_cleans_up_if_task_creation_fails(
+        self, project_id, flow, monkeypatch
+    ):
+        patched_insert_raises_error = MagicMock(side_effect=Exception())
+        monkeypatch.setattr(
+            "prefect.models.Task.insert_many", patched_insert_raises_error
+        )
+        patched_delete = MagicMock()
+        monkeypatch.setattr("prefect_server.api.flows.delete_flow", patched_delete)
+        with pytest.raises(Exception):
+            flow_id = await api.flows.create_flow(
+                project_id=project_id, serialized_flow=flow.serialize()
+            )
+            assert patched_delete.called
+
+    async def test_create_flow_cleans_up_if_edge_creation_fails(
```

Comment:
These seems like a good case for a parametrized test on `"Edge"` and `"Task"` but I do not feel strongly about it.

Reply:
might as well use this opportunity to get more familiar with pytest parameterization!